src/Workspaces/Core/Portable/Simplification/AbstractSimplificationService.cs

```diff
@@ -102,27 +104,22 @@ protected virtual SyntaxNode TransformReducedNode(SyntaxNode reducedNode, Syntax
         // Create a simple interval tree for simplification spans.
         var spansTree = new TextSpanIntervalTree(spans);
 
-        bool isNodeOrTokenOutsideSimplifySpans(SyntaxNodeOrToken nodeOrToken)
-            => !spansTree.HasIntervalThatOverlapsWith(nodeOrToken.FullSpan.Start, nodeOrToken.FullSpan.Length);
-
-        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-        var root = await semanticModel.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+        var root = (TCompilationUnitSyntax)await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
         // prep namespace imports marked for simplification 
         var removeIfUnusedAnnotation = new SyntaxAnnotation();
         var originalRoot = root;
-        root = PrepareNamespaceImportsForRemovalIfUnused(document, root, removeIfUnusedAnnotation, isNodeOrTokenOutsideSimplifySpans);
+        root = PrepareNamespaceImportsForRemovalIfUnused(root, removeIfUnusedAnnotation, IsNodeOrTokenOutsideSimplifySpans);
         var hasImportsToSimplify = root != originalRoot;
 
         if (hasImportsToSimplify)
         {
             document = document.WithSyntaxRoot(root);
-            semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-            root = await semanticModel.SyntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
+            root = (TCompilationUnitSyntax)await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
>root =  [](http://example.com/codeflow?start=12&length=7)

dumb question, this does actually change the value of root, right?

Reply:
yes.   when we set the root, we get a new *syntax tree* with the root parented by that (that involves a fork of the red node, but it reuses the underlying green node).